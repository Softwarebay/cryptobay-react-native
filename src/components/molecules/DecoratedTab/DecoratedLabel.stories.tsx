import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { DecoratedTab } from './DecoratedTab';

storiesOf('Molecules/DecoratedTab', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Unchecked Decorated Tab', () => (
    <DecoratedTab
      label={text('label', 'Collectibles')}
      checked={false}
      onPress={action('onPress')}
    />
  ))
  .add('Checked Decorated Tab', () => (
    <DecoratedTab
      label={text('label', 'Token')}
      checked={true}
      onPress={action('onPress')}
    />
  ));
