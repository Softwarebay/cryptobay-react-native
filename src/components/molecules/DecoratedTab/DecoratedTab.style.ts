import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  decoratedTab: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 16,
  },
  checked: {
    borderBottomColor: '#FFFFFF',
    borderBottomWidth: 2,
  },
  checkedLabel: {
    color: '#FFFFFF',
  },
  uncheckedLabel: {
    color: '#888DAA',
  },
});
