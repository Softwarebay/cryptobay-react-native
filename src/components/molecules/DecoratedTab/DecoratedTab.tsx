import React from 'react';
import { ViewStyle } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';

import { Label } from '../../atoms';

import { styles } from './DecoratedTab.style';

type DecoratedTabProps = {
  label: string;
  checked: boolean;
  onPress: () => void;
  style?: ViewStyle | ViewStyle[];
  testID?: string;
  labelTestID?: string;
};

export const DecoratedTab: React.FC<DecoratedTabProps> = ({
  label,
  checked,
  onPress,
  style,
  testID,
  labelTestID,
}) => {
  const handleStyleByChecked = checked && styles.checked;
  const handleLabelStylesByChecked = checked
    ? styles.checkedLabel
    : styles.uncheckedLabel;

  return (
    <TouchableOpacity
      style={[style, styles.decoratedTab, handleStyleByChecked]}
      testID={testID}
      onPress={onPress}>
      <Label
        value={label}
        style={handleLabelStylesByChecked}
        testID={labelTestID}
      />
    </TouchableOpacity>
  );
};
