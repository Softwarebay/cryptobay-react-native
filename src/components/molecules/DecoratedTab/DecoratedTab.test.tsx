import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { DecoratedTab } from './DecoratedTab';

const mockOnPress = jest.fn();

describe('DecoratedTab Molecule', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(
      <DecoratedTab label={''} checked={false} onPress={mockOnPress} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test('ensure decorated tab is checkable', () => {
    const { rerender } = render(
      <DecoratedTab
        label="Token"
        checked={false}
        onPress={mockOnPress}
        testID="molecule-decorated-tab"
        labelTestID="decorated-tab-label"
      />,
    );
    const label = screen.getByTestId('decorated-tab-label');
    expect(label).toHaveStyle({
      color: '#888DAA',
    });
    expect(label).toHaveTextContent('Token');
    rerender(
      <DecoratedTab
        label="Token"
        checked={true}
        onPress={mockOnPress}
        testID="molecule-decorated-tab"
        labelTestID="decorated-tab-label"
      />,
    );
    expect(label).toHaveStyle({
      color: '#FFFFFF',
    });
  });

  test('ensure onPress works correctly', () => {
    render(
      <DecoratedTab
        label="Token"
        checked={false}
        onPress={mockOnPress}
        testID="molecule-decorated-tab"
      />,
    );
    fireEvent.press(screen.getByTestId('molecule-decorated-tab'));
    expect(mockOnPress).toHaveBeenCalled();
  });
});
