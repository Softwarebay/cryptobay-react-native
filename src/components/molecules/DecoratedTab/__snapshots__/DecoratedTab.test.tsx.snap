// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecoratedTab Molecule Component Snapshot 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          Object {
            "color": "#888DAA",
          },
        ]
      }
    />
  </View>
</RNGestureHandlerButton>
`;
